### DISCUSSION TITLE
Java  OTel Agent

### DISCUSSION THREAD

**Discussion by suman2007** [open-telemetry/opentelemetry-java-instrumentation#10513] [Q&A]
1. Is there any extra thread or process created for OTel Java Agent along with the main service thread?
2. How do we determine the list of libraries being used during Autoinstrumentation?

**Comment by laurit** [comment]
> Is there any extra thread or process created for OTel Java Agent along with the main service thread?

otel java agent does not create extra processes, but it does create extra threads

> How do we determine the list of libraries being used during Autoinstrumentation?

I'm not sure what you mean by this, please elaborate.

**Reply by suman2007** [reply]
> How do we determine the list of libraries being used during Autoinstrumentation?

When I run my service with java agent with the command **java -javaagent:opentelemetry-javaagent-all.jar -jar service.jar**
is there any way to identify which library or package gets instrumented in bytecode? For example like any other javaAgent.
I need to identify the list of packages that were directly injected inside my service class.

> Is there any extra thread or process created for OTel Java Agent along with the main service thread?

Could it be possible to share the list od thread or is any way to find it.
